core/src/options/input_variant.rs

```diff
@@ -95,6 +98,12 @@ impl ParseAttribute for InputVariant {
             }
 
             self.skip = FromMeta::from_meta(mi)?;
+        } else if path.is_ident("word") && self.data.is_unit() {
```

Comment:
I'd suggest moving the second half of this check down; we should enter this branch for `word` and have a dedicated error message if the variant has fields.